src/components/EditorAndOutput.vue

```diff
@@ -213,6 +221,13 @@ function useOpenFile(startFile: FilePath | null, fs: ReactiveFilesystem) {
     });
   });
 
+  const isReadonly = computed(() => {
+    if (keyedCode.value === null) {
+      return true;
+    }
+    return keyedCode.value.name.endsWith(".graph.wgsl");
+  });
```

Comment:
Every node graph now comes with a generated `.graph.wgsl` file. Since those are auto-generated, it doesn't make sense to edit them.
So, they're readonly.